webassembly/app.html

```diff
@@ -223,6 +221,57 @@ <h1 class="title">F3D Web</h1>
         // do a first render and start the interactor
         Module.engineInstance.getWindow().render();
         Module.engineInstance.getInteractor().start();
+
+        // Parse mesh from url-param search query hash
+        function filename_for_model_url(model_url, extension_parsed, contentDisposition) {
+          // Build filename given extension urlparam or response header content-disposition
+          let filename;
+          if (extension_parsed) {
+            filename = `model_urlparam.${extension_parsed}`;
+          } else {
+            // If extension is not provided by user, try to get it auto from content-disposition header of url extension 
+            if (contentDisposition) {
+              filename = contentDisposition.split('filename=')[1].split(';')[0];
+            } else {
+              try {
+                filename = model_url.split('/').pop();
+                const url_extension = '.' + model_url.split('.').at(-1) ;
+              } catch {
+                throw new Error(`Could not infer file extension from content-disposition or url extension. Please enforce it via the extension urlparam`);
+              }
+            }
+          }
+          return filename;
```

Comment:
oops, submitted the previous remark as a single comment instead of adding to a review, my bad!
```suggestion
          throw ...
```
You can throw here unconditionally, if we hit a good code path we would have returned already. That also ensure the function can never return an undefined `filename`

Reply:
Ah true, just removed the try/catch and throw blocks. Thanks again!

Reply:
`filename` doesn't exist anymore so you need to remove this last `return`